src/logger.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crocksdb_ffi;
+use libc::{c_char, c_void};
+use librocksdb_sys::DBLogger;
+use std::ffi::VaList;
+
+pub trait Logger: Send + Sync {
```

Comment:
We also need the other way around - 
* Add C API for rocksdb `NewEnvLogger`, which returns a `crocksdb_logger_t`: https://github.com/facebook/rocksdb/blob/2d9efc9ab2475876270741f0592da13eef44096b/include/rocksdb/env.h#L1603
* make an rust impl of `Logger` trait that wraps around `crocksdb_logger_t`.

Reply:
Done